In [1]:
import vicra_toolbox
import numpy as np
import csv

np.set_printoptions(suppress=True)

# RotTransMatrix_6Params

In [2]:
# METHOD 1
#    One liner: Rotation/Translation to Decomposed 6 elements

#    INPUT    : 12 Vicra elements (Vicra position 2-13)
#    Order    : R11,R12,R13,T14,R21,R22,R23,T24,R31,R32,R33,T34

#    OUTPUT   : Decomposed 6 parameters 
#    Order    : Tx,Ty,Tz,Rx,Ry,Rz

Input_elements = np.array([0.994423 , -0.104231, 0.016097 ,  0.006883, \
                           0.103476 , 0.993736 , 0.042212 , 12.938414, \
                           -0.020396, -0.040311, 0.998979 , 5.866369 ])

Output = vicra_toolbox.RotTransMatrix_6Params(Input_elements,1)
print(Output)

[ 0.006883   12.938414    5.866369   -2.41960392  0.92233064  5.9836494 ]


In [3]:
# METHOD 2
#    One liner: Decomposed 6 elements to Rotation/Translation

#    INPUT    : Decomposed 6 parameters 
#    Order    : Tx,Ty,Tz,Rx,Ry,Rz

#    OUTPUT   : 12 Vicra elements (Vicra position 2-13) 
#    Order    : R11,R12,R13,T14,R21,R22,R23,T24,R31,R32,R33,T34

Input_elements2 = np.array([0.006883, 12.938414, 5.866369, -2.41960392, 0.92233064, 5.9836494])

Output2 = vicra_toolbox.RotTransMatrix_6Params(Input_elements2,2)
print(Output2)

[ 0.99442282 -0.10423114  0.01609701  0.006883    0.10347584  0.99373583
  0.04221203 12.938414   -0.02039599 -0.04031096  0.99897899  5.866369  ]


In [4]:
# METHOD 3

Output2 = vicra_toolbox.RotTransMatrix_6Params(Input_elements2,3)

Input method is not recognized in this function. Please choose 1 or 2.
1 = Rotation/Translation to 6 elements
2 = 6 elements to Rotation/Translation


# Relative_motion_A_to_B

In [5]:
#   ** Input **
#   Frame002 is your new REFERENCE  --> input img_vc1 (14 elements)
#   Frame006 is your new MOVE FRAME --> input img_vc2 (14 elements)
#   Order : Time,R11,R12,R13,T14,R21,R22,R23,T24,R31,R32,R33,T34,uncertainty

#   ** Output **
#   One line VC format (14 elements) - relative motion from frame B to frame A
#   Order : Time,R11,R12,R13,T14,R21,R22,R23,T24,R31,R32,R33,T34,uncertainty

# Example:
# R : reference gate;
# A : moving gate1;
# B : moving gate2;

# R = img_vc1 * A;
# R = img_vc2 * B;
# => A = inv(img_vc1) * img_vc2 * B

# Output: output_matr = inv(img_vc1) * img_vc2

VC1 = './test_data/test_vc1.vc'
VC2 = './test_data/test_vc2.vc'

# read vicra
csvFile1 = open(VC1, "r")
reader1 = csv.reader(csvFile1)
for item in reader1:
    img_vc1 = np.array(item)
csvFile1.close()

csvFile2 = open(VC2, "r")
reader2 = csv.reader(csvFile2)
for item in reader2:
    img_vc2 = np.array(item)
csvFile2.close()
    
output_matr = vicra_toolbox.Relative_motion_A_to_B(img_vc1,img_vc2)
print(output_matr)

[ 0.          0.99999975 -0.00032784 -0.00060372  0.00736467  0.00032465
  0.99998897 -0.004669    0.16730146  0.00060544  0.00466861  0.99998941
  0.39746919  1.        ]


* MATLAB output \
output_matr =

         0    1.0000   -0.0003   -0.0006    0.0074    0.0003    1.0000   -0.0047    0.1673    0.0006    0.0047    1.0000    0.3975    1.0000

# DL_HMC_concat_VC

In [6]:
#   ** Input **
#   img_vc1  : 1 line of VC, 14 elements
#   img_vc2  : 1 line of VC, 14 elements
#   Order    : Time,R11,R12,R13,T14,R21,R22,R23,T24,R31,R32,R33,T34,uncertainty

#   ** Output **
#   MOLAR_VC_matrix_full : resulting transformation matrix (1 line of VC, 14 elements)
#   Order                : Time,R11,R12,R13,T14,R21,R22,R23,T24,R31,R32,R33,T34,uncertainty

#   VC_6_params          : resulting 6 parameters (translation and rotation xyz)
#   Order                : Tx,Ty,Tz,Rx,Ry,Rz

# Example:
# R : reference gate;
# A : moving gate1;
# B : moving gate2;

# R = img_vc1 * A;
# A = img_vc2 * B;
# => R = img_vc1 * img_vc2 * B

# Output: MOLAR_VC_matrix_full = img_vc1 * img_vc2

VC1 = './test_data/test_vc1.vc'
VC2 = './test_data/test_vc2.vc'

# read vicra
csvFile1 = open(VC1, "r")
reader1 = csv.reader(csvFile1)
for item in reader1:
    img_vc1 = np.array(item)
csvFile1.close()

csvFile2 = open(VC2, "r")
reader2 = csv.reader(csvFile2)
for item in reader2:
    img_vc2 = np.array(item)
csvFile2.close()

MOLAR_VC_matrix_full, decomposed_transMatrix = vicra_toolbox.DL_HMC_concat_VC(img_vc1, img_vc2)
print(MOLAR_VC_matrix_full)
print(decomposed_transMatrix)

[  0.           0.9937216    0.10049832   0.04916107   4.5508156
  -0.10308617   0.99324448   0.05327427   1.04686592  -0.04347448
  -0.05800667   0.99736863 -11.87900005   1.        ]
[  4.5508156    1.04686592 -11.87900005  -3.05751107   2.81784265
  -5.77489102]


* MATLAB output 

    MOLAR_VC_matrix_full =

    0    0.9937    0.1005    0.0492    4.5508   -0.1031    0.9932    0.0533    1.0469   -0.0435   -0.0580    0.9974  -11.8790    1.0000
    
    VC_6_params =

    4.5508    1.0469  -11.8790   -3.0575    2.8179   -5.7749